compiler-core/src/build/project_compiler.rs

```diff
@@ -112,6 +114,9 @@ where
 
     /// Returns the compiled information from the root package
     pub fn compile(&mut self) -> Result<Package> {
+        let lock = NamedLock::create("gleam-compile").expect("Could not lock build directory");
+        let _lock_guard = lock.try_lock().expect("Could not lock build directory");
```

Comment:
Can we use `lock`? That will wait for it to unlock rather than crashing the compiler if it can't lock, which is what `expect` does.

Reply:
Okay!